static/app/components/compactSelect/control.tsx

```diff
@@ -341,6 +348,11 @@ export function Control({
     },
   });
 
+  const showClearButton = useMemo(
+    () => selectedOptions.flat().length > 0,
+    [selectedOptions]
+  );
+
```

Comment:
The clear button will only be hidden if no option has been selected:
![image](https://user-images.githubusercontent.com/44172267/228975248-97b8614f-eace-47d5-b44a-bbea72d16ebf.png)